### ISSUE TITLE
Decoding in rpn

### ISSUE THREAD

**Issue by Nazarko12345** [learnml-today/object-detection-with-pytorch#20]
We encode proposals on [0,1] and then apply loss funciton  but here seems like decoding applied to unnormlized anchors : 

`proposals = self.box_coder.decode(pred_bbox_deltas.detach(), anchors)
`
Is it good to decode pred_bbox_deltas with unnormalized anchors?Or we dont do any normalization to [0,1] coordinates in RPN?

**Comment by creotiv** [comment]
1) why you think that we encode proposals on [0,1]? All coordinates params are relative, target relates to anchors and creates gt proposals, and then prediction relates to proposals, creating gt detection.

2) we do normalization through scaling in encoding and decoding (weight param)

3) we don't lock prediction in [0,1] with some sigmoid for example because this gives additional regularization for the network. Which results in less overfitting.

**Comment by Nazarko12345** [comment]
So whats the point to make reguralization to [0,1] is ssd?or we just making same moves like in paper?

**Comment by creotiv** [comment]
We don't do it in SSD either. We only limit default boxes to [0,1]
